liboptv/src/parameters.c

```diff
@@ -337,3 +337,57 @@ int compare_mm_np(mm_np *mm_np1, mm_np *mm_np2)
 		return 0;
 	return 1;
 }
+
+/* read_shaking_par() reads 4 shaking parameters from a text file line by line
+
+ Arguments:
+ char *filename - path to the text file containing the parameters.
+
+ Returns:
+ Pointer to a newly-allocated shaking_par structure. Don't forget to free
+ the new memory that is allocated for the image names.
+ Returns NULL in case of any reading error.
+ */
+shaking_par * read_shaking_par(char* filename) {
+	FILE *par_file;
+	shaking_par *ret;	//returned pointer to shaking_par struct
+
+	par_file = fopen(filename, "r");
+	if (par_file == NULL) {
+		printf("Failed to open %s\n", filename);
+		return NULL;
+	}
+
+	ret = (shaking_par *) malloc(sizeof(shaking_par));
+
+	if (	   fscanf(par_file, "%d\n", &(ret->seq_first)) 				== 0
+			|| fscanf(par_file, "%d\n", &(ret->seq_last)) 			== 0
+			|| fscanf(par_file, "%d\n", &(ret->max_shaking_points)) 	== 0
+			|| fscanf(par_file, "%d\n", &(ret->max_shaking_frames)) 	== 0 )
```

Comment:
Of course. My problem is not twith the or, it's with the goto. The goto is there to avoid nesting, but with the short-circuit 'or' you don't nest so much, so you can do a simple if/else.